pkg/lib/logging/logger.go

```diff
@@ -45,25 +46,81 @@ type Logger struct {
 	level int
 }
 
-// Get a named logger.
-func WithName(name string, kvpair ...interface{}) *Logger {
-	l := &Logger{
-		Real: Factory.New(),
-		name: name,
+type LevelLogger interface {
+	Info(msg string, kv ...interface{})
+	Enabled() bool
+	Error(err error, msg string, kv ...interface{})
+	WithValues(kv ...interface{}) LevelLogger
+	WithName(name string) LevelLogger
+	V(level int) LevelLogger
+	Trace(err error, kvpair ...interface{})
+}
+
+type levelLoggerImpl struct {
+	real  logr.Logger
+	level int
+}
+
+func (l *levelLoggerImpl) Info(msg string, kv ...interface{}) {
+	if Settings.allowed(l.level) {
```

Comment:
Info should always be logged, no?

Reply:
I took the existing the implementation, theoretically someone could want to only log errors

Reply:
yeah, ok, if Trace is used for errors then maybe someone envisioned using Info for debug logs...